website/src/components/Chat/ChatConversation.tsx

```diff
@@ -50,32 +61,156 @@ export const ChatConversation = ({ chatId }: ChatConversationProps) => {
       await new Promise(requestAnimationFrame);
     }
 
-    setMessages((old) => [...old, { ...response.assistant_message, content: responseMessage }]);
+    setMessages((old) => [...old, { ...response.assistant_message, content: responseMessage, score: 0 }]);
     setResponse(null);
   }, [chatId, messages]);
 
+  const { trigger } = useSWRMutation<
+    any,
+    any,
+    any,
+    {
+      message_id: string;
+      chat_id: string;
+      score: number;
+    }
+  >(API_ROUTES.CHAT_MESSAGE_VOTE, post);
+
+  const handleOnVote: ChatMessageEntryProps["onVote"] = useCallback(
+    async ({ chatId, messageId, newScore, oldScore }) => {
+      // immediately set score
+      setMessages((messages) =>
+        messages.map((message) => {
+          return message.id !== messageId ? message : { ...message, score: newScore };
+        })
+      );
+      try {
+        await trigger({ chat_id: chatId, message_id: messageId, score: newScore });
+      } catch {
+        // TODO maybe we should trigger a toast or something
+        // revert on any error
+        setMessages((messages) =>
+          messages.map((message) => {
+            return message.id !== messageId ? message : { ...message, score: oldScore };
+          })
+        );
+      }
```

Comment:
I do understand why this is here, however, we don't handle any errors for almost all of our API requests.

I don't think this is necessary.

Reply:
But we are calculating the score base on the current score. It might be buggy without handling errors properly